sklearn/decomposition/dict_learning.py

```diff
@@ -309,19 +309,45 @@ def sparse_encode(X, dictionary, gram=None, cov=None, algorithm='lasso_lars',
     return code
 
 
-def _update_dict(dictionary, Y, code, verbose=False, return_r2=False,
-                 random_state=None):
-    """Update the dense dictionary factor in place.
+def _proj_l2(v, copy=False):
+    """Projects v unto l2 unit ball in-place.
```

Comment:
hm, that's what happens when docstring and func prototype follow different evolutions and become out-of-sync. Good catch. Thx :)